### PR TITLE
Add course view for teachers

### PR DESCRIPTION
**Pull request by mehalshah**
Add the course view for teachers that shows all the courses they are enrolled in. Also removed everything for individual course enrollments other than the new course enrollment form (which will eventually get replaced)

### CODE REVIEW

File: dashboard/app/controllers/application_controller.rb [hunk 1]
```diff
@@ -223,6 +223,12 @@ def require_levelbuilder_mode
     end
   end
 
+  def require_teacher
+    unless current_user.user_type == User::TYPE_TEACHER || current_user.admin
+      raise CanCan::AccessDenied.new('Teacher dashboard is restricted to teachers')
+    end
+  end
```
Comment (joshlory) [inline dashboard/app/controllers/application_controller.rb]: Can this be done as a CanCanCan Ability instead?

**Comment by joshlory** [comment]
Should there be default content for when the user has no course enrollments?
screen shot 2016-04-01 at 3 12 12 pm

**Comment by mehalshah** [comment]
As per discussion - put the logic back into user_course_enrollment for this.

File: dashboard/app/controllers/plc/user_course_enrollments_controller.rb [hunk 2]
```diff
@@ -20,17 +11,14 @@ def new
...
     end
   end
 
-  # DELETE /plc/user_course_enrollments/1
-  # DELETE /plc/user_course_enrollments/1.json
-  def destroy
-    @user_course_enrollment.destroy
-    redirect_to action: :index
+  def course_view
+    @enrollments = current_user.plc_enrollments
```
Comment (joshlory) [inline dashboard/app/controllers/plc/user_course_enrollments_controller.rb]: Prefer using the cancancan-provided scope, e.g:

``` ruby
@enrollments = @enrollments.where(user: current_user)
```

The generated SQL is the same, and this feels more rails-y.

``` sql
SELECT `plc_user_course_enrollments`.* FROM `plc_user_course_enrollments` WHERE `plc_user_course_enrollments`.`user_id` = 1
```

File: dashboard/app/controllers/plc/user_course_enrollments_controller.rb [hunk 3]
```diff
@@ -20,17 +11,14 @@ def new
   # POST /plc/user_course_enrollments.json
   def create
     if @user_course_enrollment.save
-      redirect_to @user_course_enrollment, notice: 'User course enrollment was successfully created.'
```
Comment (joshlory) [inline dashboard/app/controllers/plc/user_course_enrollments_controller.rb:23]: Worth keeping this notice?

Comment (mehalshah) [inline dashboard/app/controllers/plc/user_course_enrollments_controller.rb:23]: Eh - I didn't think so. The next iteration of the course view dashboard may include some one-time text that introduces you to the contents of the view, and that'll accomplish the same thing

**Comment by joshlory** [comment]
LGTM

**Pull request merged**